components/org.wso2.carbon.siddhi.editor.core/src/main/java/org/wso2/carbon/siddhi/editor/core/util/eventflow/EventFlow.java

```diff
@@ -0,0 +1,320 @@
+        }
+
+        // Create Stream Nodes
+        for (StreamInfo stream : siddhiAppMap.getStreams()) {
+            createNode(NodeType.STREAM, stream.getId(), stream.getName(), stream.getDefinition());
+        }
+
+        // Create Table Nodes
+        for (TableInfo table : siddhiAppMap.getTables()) {
+            createNode(NodeType.TABLE, table.getId(), table.getName(), table.getDefinition());
+        }
+
+        // Create Window Nodes
+        for (WindowInfo window : siddhiAppMap.getWindows()) {
+            createNode(NodeType.WINDOW, window.getId(), window.getName(), window.getDefinition());
+        }
+
+        // Create Aggregation Nodes
+        for (AggregationInfo aggregation : siddhiAppMap.getAggregations()) {
+            createNode(NodeType.AGGREGATION, aggregation.getId(), aggregation.getName(), aggregation.getDefinition());
+        }
+
+        // Create Function Nodes
+        for (FunctionInfo function : siddhiAppMap.getFunctions()) {
+            createNode(NodeType.FUNCTION, function.getId(), function.getName(), function.getDefinition());
+        }
+
+        // Create Query Nodes
+        for (QueryInfo query : siddhiAppMap.getQueries()) {
+            createNode(NodeType.QUERY, query.getId(), query.getName(), query.getDefinition());
+        }
+
+        // Create Partition Nodes
+        for (PartitionInfo partition : siddhiAppMap.getPartitions()) {
+            createNode(NodeType.PARTITION, partition.getId(), partition.getName(), partition.getDefinition());
+
+            // Create Nodes For The Queries Inside The Partition
+            for (QueryInfo query : partition.getQueries()) {
+                createNode(NodeType.QUERY, query.getId(), query.getName(), query.getDefinition());
+            }
+
+            // Create Nodes For The Range & Value Partition Types Inside The Partition
+            for (PartitionTypeInfo partitionType : partition.getPartitionTypes()) {
+                createNode(NodeType.PARTITION_TYPE, partitionType.getId(), partitionType.getName(),
+                        partitionType.getDefinition());
+            }
+        }
+
+        // Add All The Created Nodes To The Event Flow
+        eventFlowJSON.put("nodes", nodes);
+    }
+
+    /**
+     * Creates a JSONObject that defines a node and adds it to the nodes JSONArray.
+     *
+     * @param type        Defines the type of node (Ex: stream, table, trigger etc.)
+     * @param id          A unique value given to identify a particular node
+     * @param name        The name of the node (not always unique - this is the name that will be displayed in the UI)
+     * @param description A piece of the Siddhi code where the node (stream, table etc.) is defined
+     */
+    private void createNode(NodeType type, String id, String name, String description) {
+        JSONObject jsonObject = new JSONObject();
+        jsonObject.put("type", type.getTypeAsString());
```

Comment:
"type", "parent", "child", "id", "name" and similar strings can be present as constants

Reply:
This is already done for most situations. Will do for this as well.